homeassistant/components/geniushub/__init__.py

```diff
@@ -106,11 +138,75 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Create a Genius Hub system."""
+async def validate_input(
+    hass: HomeAssistant, hass_data_in: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys with values provided by the user.
+    """
+    hass_data = dict(hass_data_in)
+    if CONF_HOST in hass_data:
+        args = (hass_data.pop(CONF_HOST),)
+    else:
+        args = (hass_data.pop(CONF_TOKEN),)
+
+    hub_uid = hass_data.pop(CONF_MAC, None)
+
+    client = GeniusHub(*args, **hass_data, session=async_get_clientsession(hass))
+
     hass.data[DOMAIN] = {}
+    hass.data[DOMAIN]["broker"] = GeniusBroker(hass, client, hub_uid)
+
+    await client.update()
+
+    _LOGGER.debug("config_flow.py:validate_input: ", extra=hass_data)
+
+    return {"title": args}
+
 
-    kwargs = dict(config[DOMAIN])
+async def _async_import(hass: HomeAssistant, base_config: ConfigType) -> None:
+    """Import a config entry from configuration.yaml."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_IMPORT},
+        data=base_config[DOMAIN],
+    )
+    if result["type"] == FlowResultType.CREATE_ENTRY:
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.7.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Genius Hub",
+            },
+        )
+        return
+    async_create_issue(
+        hass,
+        DOMAIN,
+        f"deprecated_yaml_import_issue_{result['reason']}",
+        breaks_in_ha_version="2024.7.0",
```

Comment:
```suggestion
        breaks_in_ha_version="2024.12.0",
```

Reply:
Happy to make the change but where does the date come from?

Reply:
Done

Reply:
next release + 6 month deprecation period